src/service/app_state.py

```diff
@@ -0,0 +1,144 @@
+"""
+Application state information and retrieval functions.
+
+All functions assume that the application state has been initialized via
+calling the build_app() method.
+"""
+
+import asyncio
+import logging
+import os
+from typing import NamedTuple, Optional
+
+from fastapi import FastAPI, Request
+
+from src.service.kb_auth import KBaseAuth, KBaseUser
+
+logger = logging.getLogger(__name__)
+
+
+class AppState(NamedTuple):
+    """Holds application state."""
+
+    auth: KBaseAuth
+
+
+class RequestState(NamedTuple):
+    """Holds request specific state."""
+
+    user: Optional[KBaseUser]
+    auth_error: Optional[Exception] = None
```

Comment:
I don't understand this - if an auth error occurs an exception should be thrown and the request should end, why would you need to store the error in the request state?

Reply:
I was using it in bearer. Maybe it's unnecessary - removed it.

Reply:
If you're planning to integrate auth in the same way as CTS then it shouldn't be necessary